src/main/java/dev/sigstore/KeylessSigner.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore;
+
+import com.google.api.client.util.Preconditions;
+import com.google.common.io.Resources;
+import dev.sigstore.encryption.signers.Signer;
+import dev.sigstore.encryption.signers.Signers;
+import dev.sigstore.fulcio.client.*;
+import dev.sigstore.oidc.client.OidcClient;
+import dev.sigstore.oidc.client.OidcException;
+import dev.sigstore.oidc.client.WebOidcClient;
+import dev.sigstore.rekor.client.HashedRekordRequest;
+import dev.sigstore.rekor.client.RekorClient;
+import dev.sigstore.rekor.client.RekorVerificationException;
+import dev.sigstore.rekor.client.RekorVerifier;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.spec.InvalidKeySpecException;
+import org.conscrypt.ct.SerializationException;
+
+public class KeylessSigner {
+  private final FulcioClient fulcioClient;
+  private final FulcioVerifier fulcioVerifier;
+  private final RekorClient rekorClient;
+  private final RekorVerifier rekorVerifier;
+  private final OidcClient oidcClient;
+  private final Signer signer;
+
+  private KeylessSigner(
+      FulcioClient fulcioClient,
+      FulcioVerifier fulcioVerifier,
+      RekorClient rekorClient,
+      RekorVerifier rekorVerifier,
+      OidcClient oidcClient,
+      Signer signer) {
+    this.fulcioClient = fulcioClient;
+    this.fulcioVerifier = fulcioVerifier;
+    this.rekorClient = rekorClient;
+    this.rekorVerifier = rekorVerifier;
+    this.oidcClient = oidcClient;
+    this.signer = signer;
+  }
+
+  public static Builder builderForProd()
```

Comment:
I wonder if prod is the right wording here. Presumably users will be configuring this for either their own private sigstore infra, or one of the public providers.  Perhaps something more specific like sigstorePublicGood?  

I'm also wondering about usescases for doing something with this builder prior to building? In the tests it just doesBuilderForProd().build(), but why not cut out the middleperson?